### PR TITLE
Improve Travis

### PR DESCRIPTION
**Pull request by pascalgrimaud**
Fix https://github.com/jhipster/jhipster-dependencies/issues/22

A full complete tests, similar to what it is done in [generator-jhipster](https://github.com/jhipster/generator-jhipster)

Once it's merged, I will do another PR to confirm everything works fine: upgrade to spring-boot 1.5.19, which doesn't exist yet

### CODE REVIEW

**Comment by pascalgrimaud** [comment]
Some new ideas after this night.
I will update this PR.

**Comment by PierreBesson** [comment]
I'm not sure it's a good idea to duplicate all tests files. It will be a nightmare to keep in sync.

**Comment by pascalgrimaud** [comment]
@PierreBesson : it is what I was thinking about, this night! :-) I will change this PR, specially for that

**Comment by deepu105** [comment]
@pascalgrimaud @PierreBesson what about having the test samples and entities in a common repo and linking them to all required repo using sub-modules or by git clone?

**Comment by pascalgrimaud** [comment]
What you suggested is to move [samples](https://github.com/jhipster/generator-jhipster/tree/master/travis/samples) folder to a common repo, right ?

I like this idea, but in this case, the script [build-samples.sh](https://github.com/jhipster/generator-jhipster/blob/master/travis/build-samples.sh) won't work anymore. But I don't know if a lot of people use it ?

**Comment by deepu105** [comment]
that script can move to that repo as well I guess


Thanks & Regards,
Deepu

**Comment by PierreBesson** [comment]
@deepu105 Oh my god what you are suggesting is to create even more git repos. Are you the devil :smiling_imp: ?
I'm against it as moving tests files away from generator code is a sure way that we will never update test configurations.

But don't we have a way to import those files from the generator-jhipster npm module ?

**Comment by pascalgrimaud** [comment]
No worry guys.
Some changes, already done in another branch, but Travis failed because of that error:

```
PhantomJS 2.1.1 (Linux 0.0.0) Component Tests AuditsComponent today function  should set toDate to current date FAILED
	Expected '2018-01-01' to be '2018-1-01'.
```

Full logs: https://travis-ci.org/pascalgrimaud/jhipster-dependencies/jobs/323155734

So, I guess all our builds will failed now, because of 2017/12/31 +1 day

I'm not at home, I can't fix it right now

**Comment by deepu105** [comment]
@PierreBesson haha I don't mind having more repos if it avoids duplicating code. Also I personally dont think having more repos are any problem once you are used to it. you cant get these files via NPM as we dont bundle these and they shouldn't be bundled. So if you dont want more repo then I guess only way would be to clone the master branch of generator and use the files from there

**Comment by pascalgrimaud** [comment]
PR updated. It's far cleaner than my 1st version :laughing:

**Comment by PierreBesson** [comment]
Yes it's good but I don't understand the branch checkout. Maybe add some comments.

File: travis/install-jhipster.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+set -ev
+
+#-------------------------------------------------------------------------------
+# Choose the repo
+#-------------------------------------------------------------------------------
+export JHIPSTER_REPO=https://github.com/jhipster/generator-jhipster.git
+export JHIPSTER_BRANCH=master
```
Comment (pascalgrimaud) [inline travis/install-jhipster.sh:8]: @PierreBesson : for example, in `spring-boot-2` branch of `jhipster-dependencies`, we can put here `spring-boot-2` branch for `generator-jhipster`
So it's for testing a sync branch with one upstream. I don't know if I'm clear enough ?

Comment (deepu105) [reply travis/install-jhipster.sh:8]: Awesome

**Comment by erikkemperman** [comment]
So I wanted to try to merge the jhipster-dependencies repo into the server lib, as was suggested after the mono-repo idea on the generator was voted down. But I think it would be good to have this PR in, first, so I can work on top of it. Any chance of it landing soon? Thanks!

**Review by deepu105 — APPROVED** [review]

**Comment by deepu105** [comment]
@erikkemperman it looks good to me so @pascalgrimaud plz merge if you are ok with it

**Pull request merged**